src/client/auth/oidc.rs

```diff
@@ -351,12 +360,54 @@ async fn do_two_step_auth(
     Ok(())
 }
 
```

Comment:
I tried to make this return an Iterator to avoid allocating Vecs, but I'm not sure how to handle errors internally without collecting to `<Result<Vec<_>>`.

Reply:
Ok, I found a solution, and a deeper understanding of `impl return`. I should have remembered it's existential not universal.

Reply:
Now that you've solved it, I think `Result<Vec<&'a str>>` would actually be better :)  That'll eagerly validate the list of hosts rather than only erroring if the iteration happens to hit an invalid value, and the overhead of `Vec` might not even be more than the overhead of `Box<dyn ..>` depending on list length and `Iterator` dynamic dispatch size.

Reply:
That is true for short lists. This is sort of optimizing for large lists of allowed hosts specified by users, which is probably pretty unlikely.